tests/functional/codegen/test_abi_encode.py

```diff
@@ -116,3 +116,70 @@ def abi_encode3(x: uint256, ensure_tuple: bool, include_method_id: bool) -> Byte
     human_encoded = abi_encode(f"({human_t})", (human_tuple,))
     assert c.abi_encode(*args, True, False).hex() == human_encoded.hex()
     assert c.abi_encode(*args, True, True).hex() == (method_id + human_encoded).hex()
+
+
+def test_abi_encode_length_failing(get_contract, assert_compile_failed):
```

Comment:
what is the difference between `test_abi_encode_length_failing` and `test_abi_encode_length_failing2`?

Reply:
The only difference is in `Bytes` vs `String`. Maybe I can refactor into a single test by parametrizing the type and value?

Reply:
I have consolidated these two into a single test case with the parametrize decorator. I will mark this as resolved first!